source/common/http/conn_manager_impl.cc

```diff
@@ -2037,6 +2037,33 @@ const Network::Connection* ConnectionManagerImpl::ActiveStreamFilterBase::connec
   return parent_.connection();
 }
 
+PerConnectionObjectSharedPtr
+ConnectionManagerImpl::ActiveStreamFilterBase::createPerConnectionObject(
+    const std::string& object_name, MutableHttpConnection& mutable_connection,
+    const PerConnectionObjectCreator& creation_function) {
+
+  // Return 'nullptr' if the creation_function is 'nullptr'
```

Comment:
If I make it an ASSERT, this will be caught during runtime. By returning a nullptr I am avoiding that crash here during runtime and putting the burden on the programming logic on top of this to validate. It could use ASSERT on the return call of this function.

It depends on if we want to treat it as an programming error. There could be an use case where the user after creating the object once, might want to access the object intentionally using a null creation function.

Please let me know if you would want to still make this an ASSERT. I will make this change.

Reply:
I think ASSERT is fine here for now. I can't think of any reasonable time to pass null for the creation callback and be guaranteed of getting a null result.